### PR TITLE
feat(lessons): parse judge scores from frontmatter for similarity dedupe

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## What
Implements the TODO in `get_lesson_scores()` so the similarity deduplication path in `select_best_lesson()` can prefer higher-scored lessons over newer-unscored ones.

## How
`get_lesson_scores()` now parses three frontmatter score formats (checked in order):
1. Direct `scores:` block — per-dimension numeric values (from LLM-as-judge `llm_judge_score`)
2. Nested `quality.scores:` block — same shape, one level deeper
3. LOO-style `effectiveness:` — `average:` sub-field or direct numeric

Non-numeric score values are silently dropped (not treated as zero). `select_best_lesson()`'s existing score-preferred logic is now live instead of always falling through to mtime.

## Why
When duplicate/near-duplicate lessons exist, `newest file wins` is a weak default — it can keep a recently generated lower-quality lesson over an older lesson with better judge evidence. The similarity dedupe helper had the right policy shape but no real quality signal.

## Verification
- 14 new tests in `test_similarity_scores.py` covering all three formats plus edge cases (mixed types, malformed frontmatter, empty scores, integer-to-float coercion)
- Full suite: `uv run pytest packages/gptme-lessons-extras/tests/ -v` — 53 passed
- No regression: existing similarity tests (discovery, generate, validate) all green

### CODE REVIEW

**Comment by TimeToBuildBob** [comment]
## Addressed: Greptile review feedback

Both Greptile review comments are now fixed in `2ea34bb`:

**P1 — Boolean coercion in `_coerce_numeric_scores()`** (line 149-153): 
Added `not isinstance(value, bool)` guard. A `scores:` block mixing booleans (e.g. `is_validated: true`) with real numeric scores would have coerced `True` → `1.0` and `False` → `0.0`, skewing the average and potentially selecting the wrong lesson. Now booleans are silently dropped alongside other non-numeric values.

**P2 — Missing type annotation for `fm`** (line 177):
Added `: Dict` type annotation to fix the mypy `var-annotated` error. (The inline `import yaml` is intentional — yaml is not a dependency of the `gptme-lessons-extras` package's core path, and lazy-importing keeps it off the startup path.)

**New test**: `test_boolean_values_in_scores_are_ignored` covers the bool-coercion edge case with both `true` and `false` values alongside real numeric scores.

**CI status**: All 15 new tests pass locally. The typecheck and pre-commit failures should now be resolved (the only failure was the `var-annotated` error on `fm`).

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: packages/gptme-lessons-extras/tests/test_similarity_scores.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+        result = get_lesson_scores(info)
+        assert result == {"correctness": 0.80, "brevity": 0.95}
+
+    def test_empty_scores_block_returns_empty(self, tmp_path):
+        """A `scores:` block with no numeric children returns empty."""
+        fp = _write_lesson(
+            tmp_path,
+            "empty-scores.md",
+            """\
+            ---
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply packages/gptme-lessons-extras/tests/test_similarity_scores.py:228]: Added in `2ea34bb`: `test_boolean_values_in_scores_are_ignored` covers mixed boolean and numeric `scores:` values and locks in the `bool`-subclass-of-`int` edge case.

File: packages/gptme-lessons-extras/src/gptme_lessons_extras/utils/similarity.py [hunk 2]
```diff
@@ -144,13 +144,73 @@ def find_similar_lessons(lesson_dir: Path, threshold: float = 0.7) -> List[List[
...
+    silently ignored so a malformed single dimension doesn't poison the
+    entire selection.
+
     Returns empty dict if no scores found.
     """
-    # TODO: Parse judge scores from lesson content or metadata
-    # For now, return empty dict (implement when we have score storage)
+    import yaml
+
+    content = lesson_info.get("content", "")
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply packages/gptme-lessons-extras/src/gptme_lessons_extras/utils/similarity.py:177]: Keeping the lazy `import yaml` intentionally. `gptme_lessons_extras` should not require PyYAML on its core import path, and this helper only needs YAML when score frontmatter is actually present.

File: packages/gptme-lessons-extras/src/gptme_lessons_extras/utils/similarity.py [hunk 3]
```diff
@@ -144,13 +144,73 @@ def find_similar_lessons(lesson_dir: Path, threshold: float = 0.7) -> List[List[
     return result
 
 
+def _coerce_numeric_scores(scores_dict: Dict) -> Dict[str, float]:
+    """Extract only numeric score entries, dropping non-numeric ones."""
+    result: Dict[str, float] = {}
+    for key, value in scores_dict.items():
+        if isinstance(value, (int, float)):
+            result[str(key)] = float(value)
+    return result
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply packages/gptme-lessons-extras/src/gptme_lessons_extras/utils/similarity.py:158]: Fixed in `2ea34bb` and present in the current head (`102dae9`). `_coerce_numeric_scores()` now rejects booleans explicitly, so `true`/`false` fields no longer skew the average or lesson-selection result.
